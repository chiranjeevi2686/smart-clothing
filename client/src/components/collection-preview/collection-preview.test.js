import React from "react";
import { render, fireEvent } from '@testing-library/react';
import CollectionPreview from './collection-preview.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = {

};
const mockStore = configureStore();
describe('Collection preview component', () => {
    it('should render collection preview', async () => {
        const items = [{ id: 1, imageUrl: 'www.test.com', price: '23', name: 'test' }]
        const title = 'testTitle'
        const { asFragment, getByText } = render(<Provider store={mockStore(defaultStore)}>
            <CollectionPreview title = {title} items = {items} />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        expect(getByText('TESTTITLE')).toBeInTheDocument();
    })
})
