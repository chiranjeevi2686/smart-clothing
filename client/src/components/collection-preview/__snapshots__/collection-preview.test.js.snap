// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection preview component should render collection preview 1`] = `
<DocumentFragment>
  <div
    class="sc-iBPRYJ hYYqFQ"
  >
    <h1
      class="sc-fubCfw gLfjKe"
    >
      TESTTITLE
    </h1>
    <div
      class="sc-pFZIQ dKPBra"
    >
      <div
        class="sc-gsTCUz llOAWQ"
      >
        <div
          class="sc-hKgILt ifgiZx"
        />
        <div
          class="sc-eCssSg jFCvIw"
        >
          <span
            class="sc-jSgupP jbqeKv"
          >
            test
          </span>
          <span
            class="sc-gKsewC kIfwPX"
          >
            $23
          </span>
        </div>
        <button
          class="sc-bdfBwQ fByGxQ sc-dlfnbm iPOXsS"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
